### ISSUE TITLE
Yet another build error

### ISSUE THREAD

**Issue by BoundaryValueProblems** [JuliaOpt/NLopt.jl#143]
I tried to build NLopt v0.5.1 under Julia v1.4.1, and got the following error:
```
(@v1.4) pkg> build NLopt
   Building CMake → `~/.julia/packages/CMake/ULbyn/deps/build.log`
   Building NLopt → `~/.julia/packages/NLopt/eqN9a/deps/build.log`
┌ Error: Error building `NLopt`: 
│ ┌ Warning: platform_key() is deprecated, use platform_key_abi() from now on
│ │   caller = ip:0x0
│ └ @ Core :-1
│ ERROR: LoadError: LibraryProduct(nothing, ["libnlopt_cxx"], :libnlopt, "Prefix(/Users/xxx/.julia/packages/NLopt/eqN9a/deps/usr)") is not satisfied, cannot generate deps.jl!
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] write_deps_file(::String, ::Array{LibraryProduct,1}; verbose::Bool, isolate::Bool) at /Users/xxx/.julia/packages/BinaryProvider/GeAtj/src/Products.jl:419
│  [3] top-level scope at /Users/xxx/.julia/packages/NLopt/eqN9a/deps/build.jl:65
│  [4] include(::String) at ./client.jl:439
│  [5] top-level scope at none:5
│ in expression starting at /Users/xxx/.julia/packages/NLopt/eqN9a/deps/build.jl:65
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/Pkg/src/Operations.jl:892
```
Thanks for your help!

**Comment by BoundaryValueProblems** [comment]
@staticfloat and @stevengj ; I am wondering what is your take on this problem. Is this related to the BinaryBuilder.jl? I found the following postings, which seems quite similar to what I encountered:
https://github.com/JuliaPackaging/BinaryProvider.jl/issues/172
https://github.com/JuliaPackaging/BinaryProvider.jl/issues/194
Thanks!

**Comment by stevengj** [comment]
I plan to switch over to a JLL package soon, which should fix this (JuliaRegistries/General#14686).

**Comment by BoundaryValueProblems** [comment]
@stevengj: Thanks a lot! It works now if I add the master branch of NLopt.jl.
